geoportal/geoportailv3_geoportal/views/wms.py

```diff
@@ -136,7 +137,69 @@ def internal_proxy_wms(self):
             if restriction is None or not restriction.readwrite:
                 return HTTPUnauthorized()
 
+        param_wms = ""
+        if ((internal_wms.ogc_server.type == 'arcgis')
+            and ('export' in internal_wms.url[-10:])):
+
+            # deactivated code for debug purpose
+            if False:
+                return Response('this is another hires layer\n',
+                                headers={"Content-Type": "text/plain"})
+
+            param_dict = {
+                'map_resolution': 'dpi',
+                'transparent': 'transparent',
+                'bbox': 'bbox',
+            }
+            for param in self.request.params:
+                if param.lower() in param_dict:
+                    param_wms = param_wms + param_dict[param.lower()] + "=" + \
+                                urllib.parse.quote(
+                                self.request.params.get(param, '').encode('utf-8')
+                                ) + "&"
```

Comment:
Dealing with query string «as a string» sounds a bit error prone to me.
What about storing them as dict, and finally `urlencode(` it at the end?

Reply:
good idea for improvement, I adapted the existing code without much change

Reply:
@tonio thanks for your review,
I just look at the code, the URL generation is manual everywhere, I wonder if I should not move this refactoring into a new PR (alongside with similar requests from PR #2547